### ISSUE TITLE
NPCs finish activities if they're on a different Z level

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#38516]
Currently NPCs stop performing actions if they're located on a different Z level from the PC. This is especially noticeable with short running tasks like Sort Loot

# Steps To Reproduce

1. Position a zone / zones nearby a staircase.
2. Set NPCs to work in these zones.
3. Move up / down the stairs and wait until they finish their tasks.
4. Check zones. Depending on how quickly you moved Z level very little work will be completed.

# Expected behavior

Z levels shouldn't impact NPC behaviour as it's all simulated now.

# Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-12725-ge035f12 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Makeshift Items Mod [makeshift],
    More Survival Tools [More_Survival_Tools],
    Mythological Replicas [nw_pack],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    No Rail Stations [No_Rail_Stations],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Roadheader and other mining vehicles [Heavy miners],
    No Fungal Monsters [No_Fungi],
    Beta National Guard Camp [national_guard_camp]
]

**Comment by davidpwbrown** [comment]
Zones arent z-level aware in general.
I don't know how difficult itd be to make them scannable across z-levels, as they are cached... and it gets pretty gnarly. 

This woul dbe a bug for avatar activities on other z-levels too id imagine, if the target zone of their activity was on a different z-level.

**Comment by Funguss** [comment]
That's true. Farming doesn't cause you or followers to take the stairs, and yes, I tried sending the avatar to sort loot from the upstairs of a building to the downstairs but it doesn't happen. It's very nice that work done on a Z level continues when the avatar is above/below, though. That helps immensely.
